### PR TITLE
Retarget CompilerTestUtilities & CSharpSyntaxTests to portable

### PR DESCRIPTION
**Pull request by agocke**
/cc @jaredpar @dotnet/roslyn-compiler 

This PR is technically based on https://github.com/dotnet/roslyn/pull/8406, so that one should be merge in before this one and we'll rebase it.

### CODE REVIEW

**Comment by davkean** [comment]
Make note Update 1 isn't yet on the boxes.

**Comment by agocke** [comment]
@davkean This is into the `features/coreclr` branch, so I'm not counting on passing tests.

**Comment by jaredpar** [comment]
Would like to review but can't until the master merge is complete.  Too many files :)

**Comment by agocke** [comment]
@jaredpar Rebase done :)

File: Compilers.sln [hunk 1]
```diff
@@ -115,11 +115,11 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "CommonNetCoreReferences", "
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "RunTests", "src\Tools\Source\RunTests\RunTests.csproj", "{1A3941F1-1E1F-4EF7-8064-7729C4C2E2AA}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "CSharpCompilerTestUtilities.Desktop", "src\Compilers\Test\Utilities\CSharp.Desktop\CSharpCompilerTestUtilities.Desktop.csproj", "{4371944A-D3BA-4B5B-8285-82E5FFC6D1F9}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "CSharpCompilerTestUtilities.Desktop", "src\Compilers\Test\Utilities\CSharp.Desktop\CSharpCompilerTestUtilities.Desktop.csproj", "{7A4B2176-7BFD-4B75-A61A-E25A1FDD0A1E}"
```
Comment (jaredpar) [inline Compilers.sln]: IIRC the plan was to have a single Desktop test assembly at the end.  Once we are there will that merge with this utilities project?  Would seem odd to have a utilities project consumed by a single test project.

**Comment by jaredpar** [comment]
:+1:

File: Compilers.sln (hunk 1 above)
Comment (agocke) [inline Compilers.sln]: That's my plan. I figure once we're in that situation it'll be easier to decide on the final architecture, though.

**Pull request merged**